File: app/lib/features/profile/data/dto/update_profile_request.dart
/// PUT /users/me 요청 바디.
///
/// - [nickname]은 필수(서버 검증: 1~50자).
/// - [profileImageUrl]/[bio]는 선택. null이면 키를 생략한다(부분 갱신 의도가 아니라,
///   백엔드가 빈 문자열→null로 정규화하므로 "지우기"는 빈 문자열을 보내는 쪽을 쓴다).
///   여기서는 입력이 없을 때(null) 전송에서 제외해 불필요한 덮어쓰기를 피한다.
class UpdateProfileRequest {
  const UpdateProfileRequest({
    required this.nickname,
    this.profileImageUrl,
    this.bio,
  });

  final String nickname;
  final String? profileImageUrl;
  final String? bio;

  Map<String, dynamic> toJson() => {
        'nickname': nickname,
        if (profileImageUrl != null) 'profileImageUrl': profileImageUrl,
        if (bio != null) 'bio': bio,
      };
}

File: backend/src/main/java/com/recordapp/domain/user/dto/UpdateProfileRequest.java
package com.recordapp.domain.user.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

/**
 * 프로필 수정 요청. nickname 필수, bio 선택(nullable).
 * email은 Supabase 소유라 수정 대상이 아니다(요청 바디에 없음).
 * 프로필 이미지는 이 경로에서 다루지 않는다 — 별도 업로드 엔드포인트(POST /users/me/avatar)에서만
 * 갱신하므로 닉네임·자기소개 수정이 이미지를 덮어쓰지 않는다.
 * 소유권은 SecurityContext의 userId로만 식별하므로 바디에 id를 두지 않는다(IDOR 차단).
 */
public record UpdateProfileRequest(
		@NotBlank @Size(max = 50) String nickname,
		@Size(max = 300) String bio) {
}
